src/main/java/crystal/CrystalException.java

```diff
@@ -0,0 +1,19 @@
+package crystal;
+
+
+/**
+ * Represents the CrystalException class.
+ *
+ */
+public class CrystalException extends Exception {
```

Comment:
I like how you defined a single exception for your project. How abt putting the Excpetion in a folder, so in the future, if you make more exceptions it's easier to add and also help you're testing?